### PR TITLE
XW-4593 | jquery less modules/ads

### PR DESCRIPTION
**Pull request by hakubo**
We'll be removing jquery from mobile-wiki
this is one of PRs that makes us ready for this

### CODE REVIEW

File: app/modules/ads.js [hunk 1]
```diff
@@ -357,36 +357,39 @@ class Ads {
 	isTopLeaderboardApplicable() {
 		const hasFeaturedVideo = this.getTargetingValue('hasFeaturedVideo'),
 			isHome = this.getTargetingValue('pageType') === 'home',
-			hasPageHeader = $('.wiki-page-header').length > 0,
-			hasPortableInfobox = $('.portable-infobox').length > 0;
+			hasPageHeader = !!document.getElementsByClassName('.wiki-page-header'),
```
Comment (kvas-damian) [inline app/modules/ads.js]:
```
!![]
//true
```
Please make sure that conditions still return the same value

File: app/modules/ads.js [hunk 2]
```diff
@@ -357,36 +357,39 @@ class Ads {
...
 
 	isPrefooterApplicable(isInContentApplicable) {
 		if (this.getTargetingValue('pageType') === 'home') {
-			return $('.trending-articles').length > 0;
+			return !!document.querySelector('.trending-articles');
 		}
 
-		const numberOfSections = $('.article-content > h2').length,
-			hasArticleFooter = $('.article-footer').length > 0;
+		const numberOfSections = document.querySelector('.article-content > h2').length,
```
Comment (kvas-damian) [inline app/modules/ads.js]: Very likely, this one will be evaluated to `undefined`

File: app/modules/ads.js [hunk 3]
```diff
@@ -357,36 +357,39 @@ class Ads {
...
 		if (this.getTargetingValue('pageType') === 'home') {
-			return $('.curated-content').length > 0;
+			return !!document.querySelector('.curated-content');
 		}
 
-		const $firstSection = $('.article-content > h2').first(),
-			firstSectionTop = ($firstSection.length && $firstSection.offset().top) || 0;
+		const firstSection = document.querySelector('.article-content > h2'),
+			firstSectionTop = (
+				firstSection.length &&
```
Comment (kvas-damian) [inline app/modules/ads.js]: it will always evaluate to `undefined`

**Review by kvas-damian — CHANGES_REQUESTED** [review]
Please fix pointed problems with missing `length` attribute on `DomNode`

**Review by kvas-damian — APPROVED** [review]
Please run Ads selenium tests before merging

File: app/modules/ads.js [hunk 4]
```diff
@@ -357,36 +357,39 @@ class Ads {
...
-			return $('.curated-content').length > 0;
+			return !!document.querySelector('.curated-content');
 		}
 
-		const $firstSection = $('.article-content > h2').first(),
-			firstSectionTop = ($firstSection.length && $firstSection.offset().top) || 0;
+		const firstSection = document.querySelector('.article-content > h2'),
+			firstSectionTop = (
+				firstSection &&
+				firstSection.getBoundingClientRect().top + document.body.scrollTop
```
Comment (latata) [inline app/modules/ads.js:374]: I've extracted offset to module function in `XW-4591`, we could use it here.

**Pull request merged**